druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/server/lookup/namespace/RocksDBExtractionNamespaceCacheFactory.java

```diff
@@ -69,70 +74,31 @@ public String call() {
         };
     }
 
+    public void tryResetRunnerOrLog(RocksDBExtractionNamespace extractionNamespace) {
+        try {
+            if (!extractionNamespace.cacheActionRunner.isEmpty()) {
+                cacheActionRunner = CacheActionRunner.class.cast(
+                        Class.forName(extractionNamespace.cacheActionRunner).newInstance());
+                LOG.debug("Populated a new CacheActionRunner with description " + cacheActionRunner.toString());
+            }
+        } catch(Exception e){
+            LOG.error("Failed to get a valid cacheActionRunner.", e);
+        }
+    }
+
     @Override
     public void updateCache(final RocksDBExtractionNamespace extractionNamespace,
                             final Map<String, String> cache, final String key, final byte[] value) {
-        if (extractionNamespace.isCacheEnabled()) {
-            try {
-
-                Parser<Message> parser = protobufSchemaFactory.getProtobufParser(extractionNamespace.getNamespace());
-                Descriptors.Descriptor descriptor = protobufSchemaFactory.getProtobufDescriptor(extractionNamespace.getNamespace());
-                Descriptors.FieldDescriptor field = descriptor.findFieldByName(extractionNamespace.getTsColumn());
-
-                Message newMessage = parser.parseFrom(value);
-                Long newLastUpdated = Long.valueOf(newMessage.getField(field).toString());
-
-                final RocksDB db = rocksDBManager.getDB(extractionNamespace.getNamespace());
-                if (db != null) {
-                    byte[] cacheValue = db.get(key.getBytes());
-                    if(cacheValue != null) {
-
-                        Message messageInDB = parser.parseFrom(cacheValue);
-                        Long lastUpdatedInDB = Long.valueOf(messageInDB.getField(field).toString());
+        tryResetRunnerOrLog(extractionNamespace);
```

Comment:
Everytime updateCache is called, the runner will be instantiated if it's defined. Can we add a check to do it only once in a running process?